registry/storage/driver/neofs/neofs.go

```diff
@@ -511,26 +510,20 @@ func (d *driver) Reader(ctx context.Context, path string, offset int64) (io.Read
 
 	addr := d.objectAddress(id)
 
-	p := new(client.ObjectHeaderParams).WithAddress(addr)
-	obj, err := d.sdkPool.GetObjectHeader(ctx, p)
+	obj, err := d.sdkPool.HeadObject(ctx, *addr)
 	if err != nil {
 		return nil, fmt.Errorf("couldn't head object '%s', id '%s': %w", path, id, err)
 	}
 
-	rng := object.NewRange()
-	rng.SetOffset(uint64(offset))
-	rng.SetLength(uint64(int64(obj.PayloadSize()) - offset))
+	length := uint64(int64(obj.PayloadSize()) - offset)
```

Comment:
```suggestion
	length := obj.PayloadSize() - uint64(offset)
```
Btw don't we need to check overflow case?
```
uint64(offset) >= obj.PayloadSize { ... }
```